test/sandbox/contract/tokenAddressToIdMap.ts

```diff
@@ -0,0 +1,26 @@
+import { batchExchangeContract } from 'contracts'
+import Logger from 'helpers/Logger'
+import { tokenList } from 'index'
+
+require('dotenv').config()
+
+/**
+ *  SANDBOX: Get token id from token addresses
+ *  RUN:     yarn sandbox test/sandbox/contract/tokenAddressToIdMap.ts
+ */
+const log = new Logger('sandbox:contract/tokenAddressToIdMap')
+
+async function exec (): Promise<void> {
+  const networkId = 4
+  log.info('Get token address for network id: %d', networkId)
+  const getAllTokenIds = tokenList.map(async token => {
```

Comment:
from what I remember you can't await in a `map` @Velenir help me out here

So you should do:
```js
const getPromisedAllTokenIds = tokenList.map(token => {
    const { symbol, name, addressByNetwork } = token
    const tokenAddress = addressByNetwork[networkId]
    
     return batchExchangeContract.methods.tokenAddressToIdMap(tokenAddress).call().then(tokenId => log.info(`${name} (${symbol}): ${tokenAddress}  ---->  ${tokenId}`))
})

return Promise.all(getPromisedAllTokenIds)
```

Reply:
`Array.map(async func) = Array<Promise>` You can await inside as long as you expect a promise in the end

Reply:
gotcha - efficiency wise are they the same? the `promise.all` vs `async map`?

Reply:
(in this context)

Reply:
There's no `vs` you still have to `Promise.all` them, like happens on L23

Reply:
i mean promise all-ing outside the loop

Reply:
yeah ok nvm! thanks